openslides/users/views.py

```diff
@@ -471,7 +471,51 @@ def destroy(self, request, *args, **kwargs):
 # Special API views
 
 
-class UserLoginView(APIView):
+class WhoAmIDataView(APIView):
+    def get_whoami_data(self):
+        """
+        Appends the user id to the context. Uses None for the anonymous
+        user. Appends also a flag if guest users are enabled in the config.
+        Appends also the serialized user if available.
+        """
+        user_id = self.request.user.pk or 0
+        guest_enabled = anonymous_is_enabled()
+
+        if user_id:
+            user_data = async_to_sync(element_cache.get_element_restricted_data)(
+                user_id, self.request.user.get_collection_string(), user_id
+            )
+            group_ids = user_data["groups_id"] or [GROUP_DEFAULT_PK]
+        else:
+            user_data = None
+            print(guest_enabled)
+            group_ids = [GROUP_DEFAULT_PK] if guest_enabled else []
+        print(group_ids)
+
+        # collect all permissions
+        permissions = set()
+        group_collection_string = "users/group"
+        for group_id in group_ids:
+            group = async_to_sync(element_cache.get_element_full_data)(
+                group_collection_string, group_id
```

Comment:
This should be fast, but it could be faster to get all data at the beginning of this function in one request https://github.com/OpenSlides/OpenSlides/blob/70191ce455e80d5de6069c4584443987c09fa0ce/openslides/utils/cache.py#L171

Reply:
Please test again